Pointers_Excercises/Excercise7/Source.cpp

```diff
@@ -0,0 +1,38 @@
+/*Exercise 7 - Pointer validity
+Declare the following function:
+void incrementNumber(int* numberPtr);
+The function should increase the value of whatever integer the pointer is pointing to.
+In the main function, create an integer and assign it a value. Create an int pointer and initialize it
+to be a null pointer. Then, assign the memory address of your integer to it. Pass in your pointer
+to the incrementNumber function and print the result to the screen. Run your program.
+After making sure the function is working fine, comment out the line where you are initializing
+the pointer to the memory address of your int.
+Depending on whether you validated the pointer or not, your program may have crashed. This is
+one of the dangers of pointers.
+Accessing null pointers will crash your application, so it’s important to be sure to add validity
+checks when there is even the slightest possibility that the value of a pointer will be null!*/
+
+#include <iostream>
+
+void IncrementNumber(int* NumberPtr)
+{
+	if (NumberPtr != nullptr)
```

Comment:
This is not wrong, however when dealing with pointers i think it's safe to say we can now use just the bool checker for them, so instead of the NumberPtr != nullptr, it's well known that this can also be checked by doing simply if (NumberPtr)

Reply:
Yep. You can do it both ways. It's more common to see the `if (NumberPtr)` alone, but however you prefer to use it is fine.